compiler/src/iree/compiler/Codegen/LLVMCPU/KernelDispatch.cpp

```diff
@@ -334,6 +334,31 @@ static bool x86TransposeLoweringPrecondition(linalg::GenericOp genericOp) {
            indexingMaps[1].isIdentity()));
 }
 
+/// Returns true if the `genericOp` implementing a transposition is supported by
+/// AArch64 SME, i.e. all of the following are true:
+///
+///   1. The op has 2 dimensions.
+///   2. The op has a single input and a single output.
+///   3. One of the `indexing_maps` is a permutation and the other an identity.
+static bool
+transposeLoweringPreconditionAArch64SME(linalg::GenericOp genericOp) {
+  // Check op has 2 dimensions.
+  if (genericOp.getNumLoops() != 2)
+    return false;
+
+  // Check op has single input and output.
+  if ((genericOp.getNumDpsInputs() != 1) || (genericOp.getNumDpsInits() != 1))
+    return false;
+
+  // Check all iterators are parallel.
+  if (genericOp.getNumParallelLoops() != genericOp.getNumLoops())
+    return false;
+
```

Comment:
I think this also needs to check the body of the generic is simply a `scf.yield %in`. 
I tried: 
```
linalg.generic {indexing_maps = [affine_map<(d0, d1) -> (d1, d0)>, affine_map<(d0, d1) -> (d0, d1)>], iterator_types = ["parallel", "parallel"]} ins(%2 : tensor<32x32xf32>) outs(%3 : tensor<32x32xf32>) {
^bb0(%in: f32, %out: f32):
  %mul = arith.mulf %in, %in : f32
  linalg.yield %mul : f32
} -> tensor<32x32xf32>
```

And that's still a 'transpose'  and it adds the scalable flags and you get: 
```
%11 = arith.mulf %9, %9 : vector<[4]x[4]xf32>
%12 = vector.transpose %11, [1, 0] : vector<[4]x[4]xf32> to vector<[4]x[4]xf32>
%13 = vector.transfer_write %12, %extracted_slice_0[%c0, %c0], %10 {in_bounds = [true, true]} : vector<[4]x[4]xf32>, tensor<?x?xf32>
```

Reply:
(`opKnownToSupport2DScalableVectorizationWithArmSME` would also need to check this)

Reply:
good spot! I've updated to check for this, but had to move this function to utils to also call it from `opKnownToSupport2DScalableVectorizationWithArmSME`.